pkg/kubelet/cm/helpers_linux.go

```diff
@@ -102,6 +102,18 @@ func ResourceConfigForPod(pod *v1.Pod) *ResourceConfig {
 		}
 	}
 
+	// For cpuLimits, need to compare the normal containers' resource
+	// requirement against the init-containers' and choose a bigger one.
+	// Otherwise, if a bigger init-container for a pod is provided,
+	// pod will fail to start due to "invalid argument" error when trying
+	// to write the cpu configuration into cpu.cfs_quota_us.
+	for _, container := range pod.Spec.InitContainers {
+		limits := container.Resources.Limits.Cpu().MilliValue()
+		if limits > cpuLimits {
```

Comment:
I think all the container's cgroup folder will be created under the pod sandbox cgroup folder, therefore, all the init-container and regular container shall share the same cgroup settings. IIUC, the cpuLimits here will config the settings for the pod sandbox cgroup. As a result, if we do not config the regular container's resource limitation, both init-container and regular container shall be unlimited.